### ISSUE TITLE
Async operations in component initialization

### ISSUE THREAD

**Issue by mdh-cc** [strongloop/loopback-boot#258]
I'm currently writing a loopback-component which has some async calls (database) during its initialization phase.

Unfortunatly, the component hook calls of loopback-boot do not inject a callback method like 'done' to ensure the component has finished loading before starting the next component in line (https://github.com/strongloop/loopback-boot/blob/master/lib/plugins/component.js#L50).

I would suggest the possibility to add a 'done' parameter to the exported component function like it is already done for boot scripts.

**Comment by raymondfeng** [comment]
@mdh-cc Do you want to submit a patch?

**Comment by mdh-cc** [comment]
@raymondfeng I created a pull request but had some issues with the test structure. I hope this fits all conditions.

**Comment by bajtos** [comment]
In October 2018, LoopBack 4.0 GA was released. LoopBack 3 and its supporting modules have entered LTS mode. Per our [Long Term Support Policy](https://loopback.io/doc/en/contrib/Long-term-support.html), this means new features are no longer accepted.

I am closing this GitHub issue as "wont fix".

@mdh-cc I see that you have opened a pull request almost a year ago to contribute the implementation yourself. I sincerely apologize that we were not able to review and accept your changes in time 😢
